modules/renter/hostdb/hostweight.go

```diff
@@ -366,6 +362,22 @@ func (hdb *HostDB) uptimeAdjustments(entry modules.HostDBEntry) float64 {
 	return math.Pow(uptimeRatio, exp)
 }
 
+// decayHostUpOrDowntime decays a host's historic uptime or historic downtime.
+// It also adds the new block height to the historic uptime or historic downtime.
+func decayUptimeOrDowntime(entry *modules.HostDBEntry, scan modules.HostDBScan, recentScan modules.HostDBScan) {
+	blocksPassed := scan.BlockHeight - recentScan.BlockHeight
+	timePassed := time.Duration(blocksPassed) * 10 * time.Minute
+	decay := time.Duration(math.Pow(0.5, float64(timePassed)/float64(halftimeUpDownTime)))
+
+	if recentScan.Success {
+		entry.HistoricUptime *= decay
+		entry.HistoricUptime += timePassed * decay
+	} else {
+		entry.HistoricDowntime *= decay
+		entry.HistoricDowntime += timePassed * decay
+	}
```

Comment:
you need to decay both the historic uptime and the historic downtime any time that you apply a decay. The decayed time passed should be added only to the relevant field. The result should look something like this:

```go
entry.HistoricUptime *= decay
entry.HistoricDowntime *= decay
if recentScan.Success {
    entry.HistoricUptime += timePassed*decay
} else {
    entry.HistoricDowntime += timePassed*decay
}
```